engine/table/src/main/java/io/deephaven/engine/table/impl/AddOnlyToBlinkTableAdapter.java

```diff
@@ -0,0 +1,124 @@
+     * the passed in table is {@link Table#isRefreshing() refreshing} and add only.
+     *
+     * @param table An add only table
+     * @return A blink table based on the input table
+     */
+    public static Table toBlink(@NotNull final Table table) {
+        if (!table.isRefreshing()) {
+            throw new IllegalArgumentException("Input table is not refreshing");
+        }
+
+        if (BlinkTableTools.isBlink(table)) {
+            return table;
+        }
+
+        if (!Boolean.TRUE.equals(table.getAttribute(Table.ADD_ONLY_TABLE_ATTRIBUTE))) {
+            throw new IllegalArgumentException("Argument table is not Add Only");
+        }
+
+        final MutableObject<QueryTable> resultHolder = new MutableObject<>();
+        final MutableObject<AppendToBlinkListener> listenerHolder = new MutableObject<>();
+        final BaseTable<?> coalesced = (BaseTable<?>) table.coalesce();
+        final SwapListener swapListener = coalesced.createSwapListenerIfRefreshing(SwapListener::new);
+
+        ConstructSnapshot.callDataSnapshotFunction("addOnlyToBlink", swapListener.makeSnapshotControl(),
+                (final boolean usePrev, final long beforeClockValue) -> {
+                    // Start with the same rows as the original table
+                    final TrackingRowSet resultRowSet = usePrev
+                            ? table.getRowSet().copyPrev().toTracking()
+                            : table.getRowSet().copy().toTracking();
+                    final QueryTable result = new QueryTable(resultRowSet, table.getColumnSourceMap());
+                    result.setRefreshing(true);
+                    result.setAttribute(BLINK_TABLE_ATTRIBUTE, true);
+
+                    final ListenerRecorder recorder =
+                            new ListenerRecorder("AppendToBlinkListenerRecorder", table, result);
+                    final AppendToBlinkListener listener = new AppendToBlinkListener(recorder, result);
+                    recorder.setMergedListener(listener);
+                    result.addParentReference(listener);
+                    swapListener.setListenerAndResult(recorder, result);
+
+                    listenerHolder.setValue(listener);
+                    resultHolder.setValue(result);
+                    return true;
+                });
+        listenerHolder.getValue().getUpdateGraph().addSource(listenerHolder.getValue());
+        return resultHolder.getValue();
+    }
+
+    private static final class AppendToBlinkListener extends MergedListener implements Runnable {
+        private final ListenerRecorder sourceRecorder;
+
+        private AppendToBlinkListener(@NotNull final ListenerRecorder recorder,
+                @NotNull final QueryTable result) {
+            super(Collections.singleton(recorder), Collections.emptyList(), "AppendToBlinkListener", result);
+            this.sourceRecorder = recorder;
+        }
```

Comment:
The new immutable List constructors are cooler, as is consistent naming:
```suggestion
    private static final class AppendToBlinkListener extends MergedListener implements Runnable {

        private final ListenerRecorder sourceRecorder;

        private AppendToBlinkListener(
                @NotNull final ListenerRecorder sourceRecorder,
                @NotNull final QueryTable result) {
            super(List.of(sourceRecorder), List.of(), "AddOnlyToBlinkListener", result);
            this.sourceRecorder = sourceRecorder;
        }
```